file_browser/src/components/AddDIalog.tsx

```diff
@@ -0,0 +1,112 @@
+import { useState } from "react";
+import { SystemItem } from "../classes/SystemItem";
+import { FileClass } from "../classes/File";
+import { Folder } from "../classes/Folder";
+import { observer } from "mobx-react-lite";
+import fileSystemStore from "../stores/FileSystemStore";
+
+const options = ["File", "Folder"];
+
+const AddDialog = observer(({
+  onClose
+}: {
+  onClose: (state: boolean) => void;
+}) => {
+
+  enum Selected {
+    FileOption = 0,
+    FolderOption = 1
+  }
+
+  const [selected, setSelected] = useState<number>(Selected.FileOption);
+  const [name, setName] = useState("");
+  const [content, setContent] = useState("");
+  const [error, setError] = useState("");
+
+  const handleAdd = (item: SystemItem) => {
+    (fileSystemStore.selectedItem as Folder).add(item);
+  };
+
+  const onChange = (i: any) => {
+    setSelected((prev: any) => (i === prev ? null : i));
+    setError("")
+  }
```

Comment:
On change of who? There are a few issues here:
1. `on` prefix is reserved for callbacks passed as props (events) - use `handle` here.
2. `onChange` is a very generic name. When you give your function a name, you need to make it clear so that the next developer doesn't have to scrol and look for references.

Rename this to `handleOptionChange`